.jenkins/cscs/env-clang-cuda.sh

```diff
@@ -14,7 +14,7 @@ export HWLOC_ROOT="${APPS_ROOT}/hwloc-2.0.3-gcc-8.3.0"
 module load daint-gpu
 module load cudatoolkit/10.2.89_3.29-7.0.2.1_3.5__g67354b4
 module load Boost/1.75.0-CrayCCE-20.11
-spack load cmake@3.17.3
+spack load cmake@3.18.6 # Necessary to handle cuda_std_17 compiler feature
```

Comment:
Do we need the comment here? Most likely we should just bump the minimum required version to 3.18, no?